### ISSUE TITLE
[in_app_purchase] Handling "nothing to restore" with restorePurchases()

### ISSUE THREAD

**Issue by ludwiktrammer** [flutter/flutter#85203]
I'm attempting to upgrade the `in_app_purchase` from 0.5.x to 1.0.x but it seems that the purchase restoration user flow that we are currently using on iOS is no longer achievable with the newer version of the plugin. I'd appreciate guidance on how to achieve our current functionality with `in_app_purchase` 1.0 or (alternatively) what is the recommended user experience for purchase restoration on iOS that is compatible with `in_app_purchase` 1.0.

Currently, our application works like this:
1. User taps "restore purchases" and the application starts showing a progress indicator.
2. Behind the scenses, `InAppPurchase.instance.queryPastPurchases()` is called.
3. When the call finishes, the progress indicator is hidden, and the return value is inspected.
4. If current purchases were returned: they are restored.
5. **If no current purchases were returned: a message is shown letting the user know that there was nothing to restore and suggesting making a new purchase.**

Now `queryPastPurchases()` has been removed. `restorePurchases()` can be used instead, but it doesn't return anything. So it seems the app won't be able to react when there was nothing to restore. The user won't get any feedback after tapping the button, which doesn't seem user-friendly at all. From the user's standpoint, it looks like that app just ignores the taps (because nothing happens), so it looks like the app is broken.

I'd like at least **one** of the following:
1. A way to know that nothing was restored by `restorePurchases()`
2. A way to know that there is nothing to restore, even before calling `restorePurchases()`
3. A recommendation for a different purchase restoration user experience on iOS that is at lest as user-friendly as our current one and compatible with `in_app_purchase` 1.0.

**Comment by salwaAlhajjaji** [comment]
Right, We really need `queryPastPurchases()`, I use `restorePurchases()` as trigger for `PurchaseDetails` stream but doesn't do enough work

currently i use in_app_purchase: ^1.0.6

**Comment by iamwaseemsh** [comment]
Hi, having same issue, no purchases are returning in testflight.

**Comment by BinayakAdhikari** [comment]
Having the same issue, restore purchase does nothing when there is no restorable items.

**Comment by RUSSCITY** [comment]
I don't understand why there is no "Issues" tab on the official github page of the plugin https://github.com/flutter/plugins/tree/main/packages/in_app_purchase/in_app_purchase 

I'm not sure if this is at all reading from some developers. 
Anyway this seems to be fixed in version 3.0.0 https://pub.dev/packages/in_app_purchase/changelog

**Comment by JackWRadford** [comment]
in_app_purchase: 3.0.7 seems to use in_app_purchase_android: 0.2.3, it should use atleast 0.2.3+2 which fixes restore purchases functionality.

**Comment by Dimilkalathiya** [comment]
@huycozy i think we can close this as completed as @RUSSCITY mentioned this issue is fixed in version 3.0.0

**Comment by huycozy** [comment]
@Dimilkalathiya Thanks for informing. Closing this as fixed.
